fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -511,6 +511,9 @@ private ScoredPlan planFilterWithInUnion(@Nonnull PlanContext planContext, @Nonn
         final ScoredPlan scoredPlan = planFilterForInJoin(planContext, inExtractor.subFilter(), true);
         if (scoredPlan != null) {
             scoredPlan.planOrderingKey = inExtractor.adjustOrdering(scoredPlan.planOrderingKey, true);
+            if (scoredPlan.planOrderingKey == null) {
```

Comment:
Cf. `allHaveOrderingKey` in `planOr`, the other caller of `PlanOrderingKey.mergedComparisonKey`.